lib/SIL/IR/OperandOwnership.cpp

```diff
@@ -127,559 +122,368 @@ SHOULD_NEVER_VISIT_INST(GetAsyncContinuation)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, OpenExistentialAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, OpenExistentialMetatype)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, PointerToAddress)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, PointerToThinFunction)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, ProjectBlockStorage)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, ProjectValueBuffer)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, RawPointerToRef)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, SelectEnumAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, SelectValue)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, StructElementAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, SwitchEnumAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, SwitchValue)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, TailAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, ThickToObjCMetatype)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, ThinFunctionToPointer)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, ThinToThickFunction)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, TupleElementAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, UncheckedAddrCast)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, UncheckedRefCastAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, UncheckedTakeEnumDataAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, UnconditionalCheckedCastAddr)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, AllocValueBuffer)
-CONSTANT_OWNERSHIP_INST(None, NonLifetimeEnding, DeallocValueBuffer)
+// Instructions that currently violate structural ownership requirements,
+// and therefore completely defeat canonicalization and optimization of any
+// OSSA value that they use.
+OPERAND_OWNERSHIP(PointerEscape, ProjectBox) // The result is a T*.
+OPERAND_OWNERSHIP(PointerEscape, ProjectExistentialBox)
+OPERAND_OWNERSHIP(PointerEscape, UncheckedOwnershipConversion)
+
+// Instructions that escape reference bits with unenforced lifetime.
+OPERAND_OWNERSHIP(BitwiseEscape, UncheckedBitwiseCast)
+OPERAND_OWNERSHIP(BitwiseEscape, ValueToBridgeObject)
+OPERAND_OWNERSHIP(BitwiseEscape, RefToRawPointer)
+OPERAND_OWNERSHIP(BitwiseEscape, UncheckedTrivialBitCast)
+// FIXME: verify that no-escape results are always trivial
+OPERAND_OWNERSHIP(BitwiseEscape, ConvertEscapeToNoEscape)
+
+// Instructions that end the lifetime of an owned value.
+OPERAND_OWNERSHIP(DestroyingConsume, AutoreleaseValue)
+OPERAND_OWNERSHIP(DestroyingConsume, DeallocBox)
+OPERAND_OWNERSHIP(DestroyingConsume, DeallocExistentialBox)
+OPERAND_OWNERSHIP(DestroyingConsume, DeallocRef)
+OPERAND_OWNERSHIP(DestroyingConsume, DestroyValue)
+OPERAND_OWNERSHIP(DestroyingConsume, EndLifetime)
+OPERAND_OWNERSHIP(DestroyingConsume, BeginCOWMutation)
+OPERAND_OWNERSHIP(DestroyingConsume, EndCOWMutation)
+OPERAND_OWNERSHIP(ForwardingConsume, MarkUninitialized)
```

Comment:
MarkUninitialized should be in the Instruction that move an owned value, no?

Reply:
`ForwardingConsume` is just an owned "move". This way it's consistently juxtaposed against `DestroyingConsume`.  Reborrow is also sort of a move. When we introduce a language-level "move" I'm not sure if we will ever allow that to apply to guaranteed values... probably not.

Reply:
Sorry. I misspoke. I meant that you segregated all ForwardingConsume instructions below. Shouldn't this be a few lines below?

Reply:
Fixed